### PR TITLE
CANS-644: Case Worker: Extend Client and Assessment authorization rules with assignment logic

### PR DESCRIPTION
**Pull request by DmitriyRudenko**
### JIRA Issue Link
- [CANS-644](https://osi-cwds.atlassian.net/browse/CANS-644)

### Tests
- [x] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Configuration changes

N/A

### Technical debt

N/A

### CODE REVIEW

File: legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java [hunk 1]
```diff
@@ -18,73 +32,105 @@
...
+  query = "SELECT COALESCE(MAX(CASE assignment_type "
+    + "                    WHEN 'P' "
+    + "                      THEN 'RW' "
+    + "                    WHEN 'S' "
+    + "                      THEN 'RW' "
+    + "                    WHEN 'R' "
+    + "                      THEN 'R' "
+    + "                    END), 'NONE') "
+    + "FROM ( "
+    + "       SELECT DISTINCT case_assignment.ASGNMNT_CD assignment_type "
```
[the remark this answers is not in the document]
Comment (LeonidMarushevskyi) [reply legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java]: Also tried to run this query on Int DB2 and it seems like it is not working with error:
[56038][-4743] ATTEMPT TO USE A FUNCTION WHEN THE APPLICATION COMPATIBILITY SETTING IS SET FOR A PREVIOUS LEVEL. SQLCODE=-4743, SQLSTATE=56038, DRIVER=4.23.42 [26501][-514] THE CURSOR SQL_CURLH200C1 IS NOT IN A PREPARED STATE. SQLCODE=-514, SQLSTATE=26501, DRIVER=4.23.42

[the remark this answers is not in the document]
Comment (DmitriyRudenko) [reply legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java]: fixed

[the remark this answers is not in the document]
Comment (LeonidMarushevskyi) [reply legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java]: After offline discussion decided not to do this premature optimization.

[the remark this answers is not in the document]
Comment (DmitriyRudenko) [reply legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java]: @LeonidMarushevskyi , LEFT JOIN is slower then INNER JOIN, I wouldn't recommend to use it.

**Pull request merged**